source/global.h

```diff
@@ -19,8 +19,9 @@
 
 
 /* Device dependend devines*/
-#define CPU_FREQUENCY       7370000UL   /* 7.37 MHz*/
-#define BAUDRATE            115200UL    /* 115,2 k*/
+#define CPU_FREQUENCY              7370000UL   /* 7.37 MHz*/
+#define PERIPHERAL_FREQUENCY    (CPU_FREQUENCY/2)
+#define BAUDRATE                 115200UL    /* 115,2 k*/
```

Comment:
Baudrate is probably to high. Set to a lower value or anything you already have tested.